### PR TITLE
[W5.11][W14-A3]Koo Chin Chye

### PR DESCRIPTION
**Pull request by waynekoo**
Added function to sort
Added insensitive find in search
Added function to find by tag separately

### CODE REVIEW

File: src/seedu/addressbook/commands/FindCommand.java [hunk 1]
```diff
@@ -50,11 +51,40 @@ public CommandResult execute() {
...
+     * Check if both set are disjoint
+     *
+     * @param firstSet
+     * @param secondSet
+     * @return true if two both set are disjoint
+     */
+    private boolean IsDisjointSet(Set<String> firstSet, Set<String> secondSet) {
+        return Collections.disjoint(toLower(firstSet), toLower(secondSet));
+    }
+
```
Comment (tanhengyeow) [inline src/seedu/addressbook/commands/FindCommand.java:71]: I see that you are converting `Set<String>` to array, working with the array and then formatting the output back into `Set<String>` again.

There is a way to simplify this. Using an enhance for loop to iterate through the `String` variables in `Set<String> strings`, you can just do `returnString.add(xxx.toLowerCase())` while iterating through `Set<String> strings`. That in my opinion would be much neater :)

Comment (waynekoo) [reply src/seedu/addressbook/commands/FindCommand.java:71]: oh yea! i can do that. Thanks for pointing it out

File: src/seedu/addressbook/commands/FindCommand.java [hunk 2]
```diff
@@ -50,11 +51,40 @@ public CommandResult execute() {
...
+     * Function to convert a set of string values to all small letters
+     *
+     * @param strings
+     * @return orignal set with all string to lowercase
+     */
+    private Set<String> toLower(Set<String> strings) {
+        String[] stringsArray = strings.toArray(new String[0]);
+        Set<String> returnString = new HashSet<>();
+
+        for (int i=0; i<stringsArray.length; ++i) {
```
Comment (kowshik-sundararajan) [inline src/seedu/addressbook/commands/FindCommand.java:82]: Minor suggestion: do keep in mind the spacing in expressions (ex: should be `i = 0`).

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -121,6 +121,14 @@ public UniqueTagList getAllTags() {
         return new UniqueTagList(allTags);
     }
 
+    /**
+     * Sort AddressBook according to alphabetical order.
+     */
+    public void SortAddressBook()
+    {
```
Comment (kowshik-sundararajan) [inline src/seedu/addressbook/data/AddressBook.java:128]: [Coding standard violation](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout): Use K&R style bracket.

**Review by kowshik-sundararajan — COMMENTED** [review]
Nicely done, good work on updating the JUnit tests as well! 👍 Suggestion for future is to submit different PRs for each feature unless they are dependent on each other. 

Some comments added. Please close the PR after reading them.

**Pull request closed without merging**